src/Catel.MVVM/Catel.MVVM.Shared/Collections/Extensions/SuspensionContextExtensions.cs

```diff
@@ -4,138 +4,232 @@
 // </copyright>
 // --------------------------------------------------------------------------------------------------------------------
 
-namespace Catel.Collections
-{
-    using System;
-    using System.Collections.Generic;
-    using System.Collections.Specialized;
 
+namespace Catel.Collections.Extensions
+{
+  using System;
+  using System.Collections.Generic;
+  using System.Collections.Specialized;
+  using System.Linq;
+
+  /// <summary>
+  /// The suspension context extensions.
+  /// </summary>
+  internal static class SuspensionContextExtensions
+  {
+    #region Methods
     /// <summary>
-    /// The suspension context extensions.
+    /// The create event args list.
     /// </summary>
-    internal static class SuspensionContextExtensions
+    /// <param name="suspensionContext">The suspension context.</param>
+    /// <typeparam name="T">The type of collection item.</typeparam>
+    /// <returns>The <see cref="ICollection{NotifyRangedCollectionChangedEventArgs}"/>.</returns>
+    public static ICollection<NotifyRangedCollectionChangedEventArgs> CreateEventArgsList<T>(this SuspensionContext<T> suspensionContext)
     {
-        #region Methods
-        /// <summary>
-        /// The create event args list.
-        /// </summary>
-        /// <param name="suspensionContext">The suspension context.</param>
-        /// <typeparam name="T">The type of collection item.</typeparam>
-        /// <returns>The <see cref="ICollection{NotifyRangedCollectionChangedEventArgs}"/>.</returns>
-        public static ICollection<NotifyRangedCollectionChangedEventArgs> CreateEventArgsList<T>(this SuspensionContext<T> suspensionContext)
-        {
-            // No suspension context is the same as None mode
-            var mode = suspensionContext?.Mode ?? SuspensionMode.None;
+      // No suspension context is the same as None mode
+      var mode = suspensionContext?.Mode ?? SuspensionMode.None;
+
+      // Fast return for no items in not None modes
+      // ReSharper disable once PossibleNullReferenceException
+      if (mode != SuspensionMode.None && suspensionContext.ChangedItems.Count == 0)
+      {
+        return new NotifyRangedCollectionChangedEventArgs[] { };
```

Comment:
Use ArrayShim.Empty<NotifyRangedCollectionChangedEventArgs>() for less allocations.